k2/csrc/fsa_algo.cc

```diff
@@ -540,4 +542,85 @@ bool TopSort(const Fsa &a, Fsa *b,
   return true;
 }
 
+std::unique_ptr<Fsa> CreateFsa(const std::vector<Arc> &arcs) {
```

Comment:
It's better to put this function in `fsa_util` or somewhere else, as it's not an algorithm, it's just a creator/constructor

Reply:
Need to reuse `DfsState`.

Reply:
We can't decide a place of code because it needs to use some variables..
As @mahsa7823 did here
https://github.com/danpovey/k2/blob/e50ad78cba68a00d264af84098acc1e85f0002fc/k2/csrc/properties.cc#L21-L29
I think if we use `DFS` approach in many algorithms, definitely we should define these structure and variables in a `util` header file, otherwise there will be so much duplicate code.

Reply:
@qindazhu  if we end up using the at DFS stuff more than twice I think it makes sense to consolidate it somehow.  I'm OK to do that in a future PR though.